python/cudf/cudf/core/column/string.py

```diff
@@ -4506,6 +4506,30 @@ def __init__(
             children=children,
         )
 
+    def __sizeof__(self):
+        n = 0
+        if len(self.base_children) == 2:
+            child0_size = (self.size + 1) * self.base_children[
+                0
+            ].dtype.itemsize
+
+            if self.offset:
+                child1_size = (
+                    self.base_children[0][-1]
```

Comment:
I don't think we can use `-1` here as `self.size` could be much smaller than the base child in this situation. I think we want `self.offset + self.size`